lua/nvim-lsp-ts-utils/inlay-hints.lua

```diff
@@ -2,99 +2,156 @@ local o = require("nvim-lsp-ts-utils.options")
 local u = require("nvim-lsp-ts-utils.utils")
 
 local api = vim.api
-
+local INLAY_HINTS_METHOD = "typescript/inlayHints"
 local M = {}
 
-M.state = {
-    enabled = false,
-    ns = api.nvim_create_namespace("ts-inlay-hints"),
-}
-
-function M.setup_autocommands()
-    u.buf_autocmd(
-        "TSLspImportOnCompletion",
-        "BufEnter,BufWinEnter,TabEnter,BufWritePost,TextChanged,TextChangedI",
-        "autocmd_fun()"
-    )
+local ns = api.nvim_create_namespace("ts-inlay-hints")
+M.ns = ns
+
+-- enabled[bufnr] = false
+M.enabled = {}
+
+local function buf_enabled(bufnr)
+    if bufnr ~= nil then
```

Comment:
Since we repeat it in a few places, could we extract the `bufnr` resolution logic into a helper function, kind of like how Neovim does it [here](https://github.com/neovim/neovim/blob/9804a2870f6f308f788f939f52958e3fbd2adaac/runtime/lua/vim/lsp.lua#L75)?